framework/include/utils/PerfGuard.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef PERFGUARD_H
+#define PERFGUARD_H
+
+#include "MooseTypes.h"
+
+// Forward Declarations
+class PerfGraph;
+
+/**
+ * Scope guard for starting and stopping timing for a node
+ *
+ * Note that the PerfGuard timing itself will take
+ * approximately 0.00015 milliseconds.
+ *
+ * That might not sound very long - but you still don't want
+ * that in the inside of tiny loops
+ */
+class PerfGuard
```

Comment:
Maybe inline these methods?

Reply:
It's a bit of a bummer to do because it means that the header will have to `#include "PerfGraph.h"`... which means so will every object in MOOSE... but the tradeoff might be worth it.